src/main/ngapp/accio/src/app/editor/editor.component.ts

```diff
@@ -15,148 +14,198 @@ import { ImageBlob } from '../ImageBlob';
+    //  Initialize canvas to scaled image width and height and mask to img. 
+    this.maskCanvas.nativeElement.width = imgWidth;
+    this.maskCanvas.nativeElement.height = imgHeight;
+    this.maskCtx = this.maskCanvas.nativeElement.getContext('2d');
+
     this.canvas.nativeElement.width = imgWidth * this.scaleFactor;
     this.canvas.nativeElement.height = imgHeight * this.scaleFactor;
-    this.ctx.scale(this.scaleFactor, this.scaleFactor); 
+    this.ctx = this.canvas.nativeElement.getContext('2d');
 
-    //  Initalize transparent black image data to use for mask.
-    this.maskImageData = new ImageData(imgWidth,  imgHeight);
-    
-    this.image.onload = () => {
-      this.ctx.drawImage(this.image, 0, 0, imgWidth, imgHeight);
-      this.hiddenCtx.drawImage(this.image, 0, 0);
-    }
-  }
+    //   Draws image non scaled on full canvas
+    this.ctx.drawImage(this.image, 0, 0);
 
-  /**  Returns the calculated scale for the image loaded. */
-  public getScaleFactor(): number {
-    return this.scaleFactor;
+    //  Sets the ImageData to be inputed by mask.directive
+    //  To change image data, just need to reinitialize page. 
+
+    //  Only gets the image data from 0,0 to the width and height of image,
+    //    not based on canvas
+    this.originalImageData = this.ctx.getImageData(0, 0, imgWidth, imgHeight);
+    this.clearCanvas();
+
+    this.drawScaledImage(this.image);
+    console.log('put imagedata')
+  }
+  
+ /**
+  *   Clears full canvas.
+  */
+  private clearCanvas() {
+    this.ctx.clearRect(0, 0, this.canvas.nativeElement.width, this.canvas.nativeElement.height);
+    this.ctx.beginPath();
   }
 
-  /**  Returns the original images data for reference in mask making. */
-  public getOriginalImageData(): ImageData {
-    return this.hiddenCtx.getImageData(0,0, this.image.width, this.image.height);
+ /**
+  *  Draws users image scaled to canvas and restores ctx.
+  *  @param image is either the mask image or image user is making a mask of.
+  */
+  private drawScaledImage(image: HTMLImageElement) {
+    this.ctx.save();
+    this.ctx.scale(this.scaleFactor, this.scaleFactor); 
+    this.ctx.drawImage(image, 0, 0, image.width, image.height);
+    this.ctx.restore();
   }
 
-  /**  Returns black transparent ImageData for single mask image. */
-  public getMaskImageData(): ImageData {
-    return this.maskImageData;
+ /** 
+  *  Sets new pixels in magenta, clears canvas of previous data 
+  *    and draws image and mask as scaled. Disables submit
+  *    on mask until the url is set. 
+  *  class @param this.disableFloodFill must equal true before called because 
+  *                               maskMmageData is being updated
+  *  class @param this.disableSubmit must equal true before called because 
+  *                            maskUrl is being updated in drawMask(). 
+  *  class @param maskPixels event Output() from mask.directive
+  *                    Gives the new pixels to add to the mask
```

Comment:
Nit: can you make the line spacing here consistent

Reply:
Yikes yes my bad